tooling/github-strategy.md

```diff
@@ -0,0 +1,92 @@
+
+Here are groups that can be used to define repository permissions ordered by process toil to join:
+
+1. Everyone behind company firewall
+1. Internal visibility​ (everyone on platform is in this group)
+1. Security group created automatically​
+1. Security group with self-join and NO forced renew after X time period​
+1. Security group with self-join and forced renew after X time period​
+1. Security group with self-initiated but requires waiting for manager permission.
+1. Security group with that requires emailing someone to manually add you
+
+## Pros and Cons of InnerSource Dedicated Environment
+
+Although InnerSource catalogs and organizations specifically for InnerSource have benefits, they should not be viewed as a place that holds “ALL” InnerSource.
+
+| Approach​ | Definition​ | Advantages​ | Disadvantages​ |
+| --- | --- | --- | --- |
+| InnerSource Catalogs | Repositories that want extra visibility self-select via topic tag. They are then harvested into a separate catalog.​ | Have some of the repos most interested in being reused in one place helps measurement & discovery of them.​ | Most users won’t troll a catalog separate from their workflow.​ There is a risk a perception is created that only repos in that catalog are innerSource.​ |
+| Dedicated Organization for InnerSource​ | Repos for a need to be shared widely are put in an org specifically for that purposeR​ | Heightens visibility for the repos in that org.​ | Most InnerSource occurs in Repositories that do not see InnerSource as focus or reason for existence.​ Requires moving code outside team’s main org, breaks links and workflows.​ |
+| Platform-wide Internal Visibility​ | GitHub provides an “internalVisibility” repo setting that enables a repo to be visible to all platform members.​ | Enables sharing at scale large enough for effective search & discovery.​ Applies to individual repos instead of orgs. ​| Requires trusting staff to only apply internalVisibility to lower risk repos with higher sharing needs. ​A repo centric approach may feel less traditional to orgs that have previously kept code within team or organization walled environments. |
+
+## Variations in how repository read access is distributed
+
+The extent to which a repository is visible to enterprise users varies between everyone in the enterprise, everyone in a GitHub organization, everyone within a GitHub team, or users given permissions to a repository individually. A GitHub team can either be a small or large group of individuals independent of organizational membership. Team memberships can be handled with GitHub’s interface or via a separate enterprise identity and permissions tracking system that has API connection to GitHub.
```

Comment:
> that has API connection to GitHub.

Is pointing to specific APIs or examples out of scope for this doc? Curious about that.

Reply:
It can be included, but perhaps it could be cut out into a separate document if it is not a pure GitHub configuration section!
The parts that involve IdP and SCIM integration, for example, are a bit outside the context of GitHub Enterprise configuration itself, and the third-party integration with the API may also appear a bit extended.